xbmc/Application.cpp

```diff
@@ -4211,9 +4213,9 @@ bool CApplication::OnMessage(CGUIMessage& message)
         CGUIMessage msg(GUI_MSG_PLAYLISTPLAYER_CHANGED, 0, 0, g_playlistPlayer.GetCurrentPlaylist(), param, item);
         g_windowManager.SendThreadMessage(msg);
         g_playlistPlayer.SetCurrentSong(m_nextPlaylistItem);
-        *m_itemCurrentFile = *item;
+        m_itemCurrentFile = item;
```

Comment:
I don't see any good reason for doing so. The objects references are all created in this block. if you look into the code you see:

CPlayList playList = g_playlistPlayer.GetPlaylist(g_playlistPlayer.GetCurrentPlaylist());
CFileItemPtr item = playList[m_nextPlaylistItem];
m_itemCurrentFile = item;

please tell me what's wrong with this.